### PR TITLE
[share_plus] Migrate Android part to Kotlin

### PR DESCRIPTION
**Pull request by vbuberen**
## Description

This is first of PRs to migrate Android parts of Plus plugins to Kotlin in order to modernise the code, provider better null-safety handling and, potentially, helping engage more contributors with Kotlin experience.

The migration is done manually, not via Java to Kotlin converter to avoid possible potential issues. I tried to not change the behavior as much as I could (however, in some places I added some additional checks) while using all the Kotlin possibilities. Everything works in a same way as it worked before, but the code is much more concise. 

With this PR we are getting to the point where we would not need `google-java-format` anymore. I suggest to replace it with a combination of KtLint + Detekt, which I would add in a separate PR.

Did a minor version change, since there were no breaking changes.

I plan to do the same migrations for the other plugins we have in Plus plugins.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: packages/share_plus/share_plus/android/src/main/kotlin/dev/fluttercommunity/plus/share/MethodCallHandler.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import java.io.IOException
+
+/** Handles the method calls for the plugin.  */
+internal class MethodCallHandler(private val share: Share) : MethodChannel.MethodCallHandler {
+
+  override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
+    when (call.method) {
+      "share" -> {
+        expectMapArguments(call)
+        // Android does not support showing the share sheet at a particular point on screen.
```
Comment (vbuberen) [inline packages/share_plus/share_plus/android/src/main/kotlin/dev/fluttercommunity/plus/share/MethodCallHandler.kt:14]: Tried to preserve all the original code comments

File: packages/share_plus/share_plus/android/src/main/kotlin/dev/fluttercommunity/plus/share/Share.kt [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+  fun share(text: String, subject: String?) {
+    val shareIntent = Intent().apply {
+      action = Intent.ACTION_SEND
+      type = "text/plain"
+      putExtra(Intent.EXTRA_TEXT, text)
+      putExtra(Intent.EXTRA_SUBJECT, subject)
+    }
+
+    val chooserIntent = Intent.createChooser(shareIntent, null /* dialog title optional */)
```
Comment (vbuberen) [inline packages/share_plus/share_plus/android/src/main/kotlin/dev/fluttercommunity/plus/share/Share.kt:51]: Tried to preserve all the original code comments.
I believe at a later point we could provide additional parameter to let users set the title for this chooser.

File: packages/share_plus/share_plus/android/src/main/kotlin/dev/fluttercommunity/plus/share/SharePlusPlugin.kt [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import io.flutter.plugin.common.MethodChannel
+
+/** Plugin method host for presenting a share sheet via Intent  */
+class SharePlusPlugin : FlutterPlugin, ActivityAware {
+  private lateinit var share: Share
+  private lateinit var methodChannel: MethodChannel
+
+  override fun onAttachedToEngine(binding: FlutterPluginBinding) {
+    methodChannel = MethodChannel(binding.binaryMessenger, CHANNEL)
+    share = Share(context = binding.applicationContext, activity = null)
```
Comment (vbuberen) [inline packages/share_plus/share_plus/android/src/main/kotlin/dev/fluttercommunity/plus/share/SharePlusPlugin.kt:16]: Explicitly specified names of parameters for better readability on Github

**Comment by vbuberen** [comment]
The failing pub.dev check is due to #655 and failing tests are caused by a flaky test in `package_info_plus` (it runs successfully on my local machine 🤔 )

**Review by piotrsed — APPROVED** [review]

**Review by miquelbeltran — APPROVED** [review]
Looks good! sorry for the late review

**Comment by vbuberen** [comment]
Thanks. Will make a few updates, since it is quite a lot of time passed and there are newer versions of Gradle, etc. available.

**Pull request merged**